robot/impl/robot_reconfigure_weak_dependencies_test.go

```diff
@@ -285,4 +291,363 @@ func TestUpdateWeakDependents(t *testing.T) {
+				API:                 weakAPI,
+				Model:               weakModel,
+				DependsOn:           []string{base1Name.String()},
+				ConvertedAttributes: &someTypeWithWeakAndStrongDepsConfig{},
+			},
+			{
+				Name:  base1Name.Name,
+				API:   base.API,
+				Model: fake.Model,
+				// We need the following `robot.Reconfigure` to call `Reconfigure` on this `base1`
+				// component. We change the `Attributes` field from the previous (nil) value to
+				// accomplish that.
+				Attributes: rutils.AttributeMap{"version": 1},
+			},
+			{
+				Name:  base2Name.Name,
+				API:   base.API,
+				Model: fake.Model,
+			},
+		},
+	}
+	test.That(t, weakCfg2.Ensure(false, logger), test.ShouldBeNil)
+	robot.Reconfigure(context.Background(), &weakCfg2)
+
+	res, err = robot.ResourceByName(weak1Name)
+	test.That(t, err, test.ShouldBeNil)
+	weak1, err = resource.AsType[*someTypeWithWeakAndStrongDeps](res)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, weak1.reconfigCount, test.ShouldEqual, 2)
+
+	// Reconfigure base2
+	// The following scenario is expected:
+	// 1) The base2 will reconfigure (which will bump the logical clock).
+	// 2) The completeConfig loop will check to see if there is a need to updateWeakDependents by
+	//    checking the logical clock and also whether there are dependencies on weak dependents.
+	//    The check will fail.
+	// 3) weak1 will not reconfigure, as base1 was not newly built or error during reconfiguration.
+	// 4) At the end of reconfiguration, weak1 will reconfigure once.
+	t.Log("Reconfigure base2")
+	weakCfg3 := config.Config{
+		Components: []resource.Config{
+			{
+				Name:                weak1Name.Name,
+				API:                 weakAPI,
+				Model:               weakModel,
+				DependsOn:           []string{base1Name.String()},
+				ConvertedAttributes: &someTypeWithWeakAndStrongDepsConfig{},
+			},
+			{
+				Name:       base1Name.Name,
+				API:        base.API,
+				Model:      fake.Model,
+				Attributes: rutils.AttributeMap{"version": 1},
+			},
+			{
+				Name:  base2Name.Name,
+				API:   base.API,
+				Model: fake.Model,
+				// We need the following `robot.Reconfigure` to call `Reconfigure` on this `base2`
+				// component. We change the `Attributes` field from the previous (nil) value to
+				// accomplish that.
+				Attributes: rutils.AttributeMap{"version": 1},
+			},
+		},
+	}
+	test.That(t, weakCfg3.Ensure(false, logger), test.ShouldBeNil)
+	robot.Reconfigure(context.Background(), &weakCfg3)
+
+	res, err = robot.ResourceByName(weak1Name)
+	test.That(t, err, test.ShouldBeNil)
+	weak1, err = resource.AsType[*someTypeWithWeakAndStrongDeps](res)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, weak1.reconfigCount, test.ShouldEqual, 3)
+
+	// check that getting dependencies for either resource does not increase reconfigCount
+	lRobot := robot.(*localRobot)
+	node, ok := lRobot.manager.resources.Node(weak1Name)
+	test.That(t, ok, test.ShouldBeTrue)
+	lRobot.getDependencies(weak1Name, node)
```

Comment:
The comment says "getting dependencies for _either resource_ does not".

I was assuming that'd be "base1" and "base2". But instead we're checking "weak1" and "base1". I can't tell if those cases have specific logic we're trying to exercise? Or if "base2" is considered redundant? But I'm not sure why it would be. Given it's different from base1 in that it is not an explicit dependency for weak1.